orderer/common/server/attestationserver.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright IBM Corp. 2017 All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package server
+
+import (
+	"runtime/debug"
+	"time"
+
+	cb "github.com/hyperledger/fabric-protos-go/common"
+	ab "github.com/hyperledger/fabric-protos-go/orderer"
+	"github.com/hyperledger/fabric/common/deliver"
+	"github.com/hyperledger/fabric/common/metrics"
+	"github.com/hyperledger/fabric/common/policies"
+	localconfig "github.com/hyperledger/fabric/orderer/common/localconfig"
+	"github.com/hyperledger/fabric/orderer/common/msgprocessor"
+	"github.com/hyperledger/fabric/orderer/common/multichannel"
+	"github.com/hyperledger/fabric/protoutil"
+	"github.com/pkg/errors"
+)
+
+type attestationserver struct {
+	dh    *deliver.Handler
+	debug *localconfig.Debug
+	*multichannel.Registrar
+}
+
+// NewServer creates an ab.AtomicBroadcastServer based on the broadcast target and ledger Reader
+func NewAttestationService(
+	r *multichannel.Registrar,
+	metricsProvider metrics.Provider,
+	debug *localconfig.Debug,
+	timeWindow time.Duration,
+	mutualTLS bool,
+	expirationCheckDisabled bool,
+) ab.BlockAttestationsServer {
+	s := &attestationserver{
+		dh:        deliver.NewHandler(deliverSupport{Registrar: r}, timeWindow, mutualTLS, deliver.NewMetrics(metricsProvider), expirationCheckDisabled),
+		debug:     debug,
+		Registrar: r,
+	}
+	return s
+}
+
+func (s *attestationserver) BlockAttestations(env *cb.Envelope, srv ab.BlockAttestations_BlockAttestationsServer) error {
```

Comment:
nit, but, the right parameter name would be `stream` and not `srv` since we're servicing a stream. We're the server here.

Reply:
updated it to strm